Added eth for javascript console

jimni1222 opened the pull request:
## Proposed changes

This PR introduces adding `eth` variable for javascript console env.

In order to support the same as the `eth` namespace API in ethereum, an `eth` variable has been added to the `web3.js` file.
The contents defined in the `Eth` object in `web3.js` were taken based on v1.10.9 of go-ethereum.
After that, the contents of `bindata.go` were updated using `go-bindata -nometadata -pkg deps -o bindata.go bignumber.js web3.js`.
`bindata.go` is a file automatically generated through go-generate.

In addition,  an `eth` module was additionally defined in `web3ext.go`, and the contents defined as `eth` module were taken based on `web3ext.go` of go-ethereum v1.10.9.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

jeongkyun-oh commented on console/jsre/deps/web3.js:
```diff
@@ -2645,6 +2646,7 @@ function Web3 (provider) {
     this._requestManager = new RequestManager(provider);
     this.currentProvider = provider;
     this.klay = new Klay(this);
+	this.eth = new Eth(this);
```
Wrong indentation?

jimni1222 replied:
I added commit to fix this :) THank yoU!

jeongkyun-oh commented on console/jsre/deps/web3.js:
```diff
@@ -5626,6 +5628,362 @@ Klay.prototype.isSyncing = function (callback) {
+var transfer = require('../transfer');
+
+var blockCall = function (args) {
+	return (utils.isString(args[0]) && args[0].indexOf('0x') === 0) ? "eth_getBlockByHash" : "eth_getBlockByNumber";
```
It looks the indentation doesn't match with the above.

jimni1222 replied:
I modified indentation from tab to 4 spaces.!

aeharvlee commented:
I built and checked the console, but there are some missing apis like `getHeader`, `getHeaderByHash`, `getHeaderByNumber`.

```
> eth.
eth._requestManager          eth.estimateGas              eth.getCoinbase              eth.getTransactionFromBlock  eth.protocolVersion
eth.accounts                 eth.filter                   eth.getCompilers             eth.getTransactionReceipt    eth.sendIBANTransaction
eth.blockNumber              eth.gasPrice                 eth.getGasPrice              eth.getUncle                 eth.sendRawTransaction
eth.call                     eth.getAccounts              eth.getHashrate              eth.getWork                  eth.sendTransaction
eth.coinbase                 eth.getBalance               eth.getMining                eth.hashrate                 eth.sign
eth.compile                  eth.getBlock                 eth.getProtocolVersion       eth.iban                     eth.signTransaction
eth.constructor              eth.getBlockNumber           eth.getStorageAt             eth.icapNamereg              eth.submitWork
eth.contract                 eth.getBlockTransactionCount eth.getSyncing               eth.isSyncing                eth.syncing
eth.defaultAccount           eth.getBlockUncleCount       eth.getTransaction           eth.mining
eth.defaultBlock             eth.getCode                  eth.getTransactionCount      eth.namereg
```

Below is the klay namespace apis in javascript console.
```
> klay.
klay.BigNumber                            klay.getBlockTransactionCount             klay.isAddress
klay._requestManager                      klay.getBlockWithConsensusInfo            klay.isChecksumAddress
klay.accountCreated                       klay.getBlockWithConsensusInfoRange       klay.isContractAccount
klay.accounts                             klay.getChainID                           klay.isIBAN
klay.blockNumber                          klay.getChainId                           klay.isParallelDBWrite
klay.call                                 klay.getCode                              klay.isSenderTxHashIndexingEnabled
klay.chainID                              klay.getCommittee                         klay.isSyncing
klay.chainId                              klay.getCommitteeSize                     klay.namereg
klay.clientVersion                        klay.getCompilers                         klay.padLeft
klay.compile                              klay.getCouncil                           klay.padRight
klay.constructor                          klay.getCouncilSize                       klay.pendingTransactions
klay.contract                             klay.getCypressCredit                     klay.protocolVersion
klay.decodeAccountKey                     klay.getDecodedAnchoringTransactionByHash klay.resend
klay.defaultAccount                       klay.getGasPrice                          klay.rewardbase
klay.defaultBlock                         klay.getHeader                            klay.sendIBANTransaction
klay.encodeAccountKey                     klay.getHeaderByHash                      klay.sendRawTransaction
klay.estimateComputationCost              klay.getHeaderByNumber                    klay.sendTransaction
klay.estimateGas                          klay.getPendingTransactions               klay.sendTransactionAsFeePayer
klay.filter                               klay.getProtocolVersion                   klay.sha3
klay.fromAscii                            klay.getRawTransaction                    klay.sign
klay.fromDecimal                          klay.getRawTransactionFromBlock           klay.signTransaction
klay.fromPeb                              klay.getRewardbase                        klay.signTransactionAsFeePayer
klay.fromUtf8                             klay.getStorageAt                         klay.submitTransaction
klay.gasPrice                             klay.getSyncing                           klay.syncing
klay.gasPriceAt                           klay.getTransaction                       klay.toAscii
klay.getAccount                           klay.getTransactionBySenderTxHash         klay.toBigNumber
klay.getAccountKey                        klay.getTransactionCount                  klay.toChecksumAddress
klay.getAccounts                          klay.getTransactionFromBlock              klay.toDecimal
klay.getBalance                           klay.getTransactionReceipt                klay.toHex
klay.getBlock                             klay.getTransactionReceiptBySenderTxHash  klay.toPeb
klay.getBlockNumber                       klay.iban                                 klay.toUtf8
klay.getBlockReceipts                     klay.icapNamereg
```

jeongkyun-oh commented on console/jsre/deps/web3.js:
```diff
@@ -6166,6 +6524,58 @@ var klay = function () {
+
+/// @returns an array of objects describing web3.eth.filter api methods
+var eth = function () {
+	var newFilterCall = function (args) {
```
Another wrong indentation ;)

jimni1222 replied:
My bad, thank you for detail review; i modified indentation again!

jimni1222 commented:
> I built ken in your repo and checked the console, but there are some missing apis like `getHeader`, `getHeaderByHash`, `getHeaderByNumber`.
> 
> ```
> > eth.
> eth._requestManager          eth.estimateGas              eth.getCoinbase              eth.getTransactionFromBlock  eth.protocolVersion
> eth.accounts                 eth.filter                   eth.getCompilers             eth.getTransactionReceipt    eth.sendIBANTransaction
> eth.blockNumber              eth.gasPrice                 eth.getGasPrice              eth.getUncle                 eth.sendRawTransaction
> eth.call                     eth.getAccounts              eth.getHashrate              eth.getWork                  eth.sendTransaction
> eth.coinbase                 eth.getBalance               eth.getMining                eth.hashrate                 eth.sign
> eth.compile                  eth.getBlock                 eth.getProtocolVersion       eth.iban                     eth.signTransaction
> eth.constructor              eth.getBlockNumber           eth.getStorageAt             eth.icapNamereg              eth.submitWork
> eth.contract                 eth.getBlockTransactionCount eth.getSyncing               eth.isSyncing                eth.syncing
> eth.defaultAccount           eth.getBlockUncleCount       eth.getTransaction           eth.mining
> eth.defaultBlock             eth.getCode                  eth.getTransactionCount      eth.namereg
> ```
> 
> Below is the klay namespace apis in javascript console.
> 
> ```
> > klay.
> klay.BigNumber                            klay.getBlockTransactionCount             klay.isAddress
> klay._requestManager                      klay.getBlockWithConsensusInfo            klay.isChecksumAddress
> klay.accountCreated                       klay.getBlockWithConsensusInfoRange       klay.isContractAccount
> klay.accounts                             klay.getChainID                           klay.isIBAN
> klay.blockNumber                          klay.getChainId                           klay.isParallelDBWrite
> klay.call                                 klay.getCode                              klay.isSenderTxHashIndexingEnabled
> klay.chainID                              klay.getCommittee                         klay.isSyncing
> klay.chainId                              klay.getCommitteeSize                     klay.namereg
> klay.clientVersion                        klay.getCompilers                         klay.padLeft
> klay.compile                              klay.getCouncil                           klay.padRight
> klay.constructor                          klay.getCouncilSize                       klay.pendingTransactions
> klay.contract                             klay.getCypressCredit                     klay.protocolVersion
> klay.decodeAccountKey                     klay.getDecodedAnchoringTransactionByHash klay.resend
> klay.defaultAccount                       klay.getGasPrice                          klay.rewardbase
> klay.defaultBlock                         klay.getHeader                            klay.sendIBANTransaction
> klay.encodeAccountKey                     klay.getHeaderByHash                      klay.sendRawTransaction
> klay.estimateComputationCost              klay.getHeaderByNumber                    klay.sendTransaction
> klay.estimateGas                          klay.getPendingTransactions               klay.sendTransactionAsFeePayer
> klay.filter                               klay.getProtocolVersion                   klay.sha3
> klay.fromAscii                            klay.getRawTransaction                    klay.sign
> klay.fromDecimal                          klay.getRawTransactionFromBlock           klay.signTransaction
> klay.fromPeb                              klay.getRewardbase                        klay.signTransactionAsFeePayer
> klay.fromUtf8                             klay.getStorageAt                         klay.submitTransaction
> klay.gasPrice                             klay.getSyncing                           klay.syncing
> klay.gasPriceAt                           klay.getTransaction                       klay.toAscii
> klay.getAccount                           klay.getTransactionBySenderTxHash         klay.toBigNumber
> klay.getAccountKey                        klay.getTransactionCount                  klay.toChecksumAddress
> klay.getAccounts                          klay.getTransactionFromBlock              klay.toDecimal
> klay.getBalance                           klay.getTransactionReceipt                klay.toHex
> klay.getBlock                             klay.getTransactionReceiptBySenderTxHash  klay.toPeb
> klay.getBlockNumber                       klay.iban                                 klay.toUtf8
> klay.getBlockReceipts                     klay.icapNamereg
> ```

Please check again. I rebased on latest dev which includes `eth` API structure, and now additional apis defined in `web3ext.go` should be there.
And i added those logic based on go-ethereum v1.10.9, so some klay apis not supprted by eth namespace.
For example, `getHeader` is defined in Klay (web3.js), but not in Eth.

aeharvlee commented:
@jimni1222 

I checked again :) Looks great to me now. Thanks for your work.
```shell
> eth.
eth._requestManager            eth.feeHistory                 eth.getCoinbase                eth.getStorageAt               eth.mining
eth.accounts                   eth.fillTransaction            eth.getCompilers               eth.getSyncing                 eth.namereg
eth.blockNumber                eth.filter                     eth.getGasPrice                eth.getTransaction             eth.pendingTransactions
eth.call                       eth.gasPrice                   eth.getHashrate                eth.getTransactionCount        eth.protocolVersion
eth.chainId                    eth.getAccounts                eth.getHeaderByHash            eth.getTransactionFromBlock    eth.resend
eth.coinbase                   eth.getBalance                 eth.getHeaderByNumber          eth.getTransactionReceipt      eth.sendIBANTransaction
eth.compile                    eth.getBlock                   eth.getMaxPriorityFeePerGas    eth.getUncle                   eth.sendRawTransaction
eth.constructor                eth.getBlockByHash             eth.getMining                  eth.getWork                    eth.sendTransaction
eth.contract                   eth.getBlockByNumber           eth.getPendingTransactions     eth.hashrate                   eth.sign
eth.createAccessList           eth.getBlockNumber             eth.getProof                   eth.iban                       eth.signTransaction
eth.defaultAccount             eth.getBlockTransactionCount   eth.getProtocolVersion         eth.icapNamereg                eth.submitTransaction
eth.defaultBlock               eth.getBlockUncleCount         eth.getRawTransaction          eth.isSyncing                  eth.submitWork
eth.estimateGas                eth.getCode                    eth.getRawTransactionFromBlock eth.maxPriorityFeePerGas       eth.syncing
```

aidan-kwon commented on console/jsre/deps/web3.js:
```diff
@@ -5626,6 +5628,362 @@ Klay.prototype.isSyncing = function (callback) {
+
+    });
+
+    var getCompilers = new Method({
+        name: 'getCompilers',
+        call: 'eth_getCompilers',
+        params: 0
+    });
```
@aeharvlee @jimni1222 Will you implement this? If not, we may deprecate this and the same one in `klay` namespace.

jimni1222 replied:
"Deprecate" means remove Method declaration here?

aeharvlee replied:
@jimni1222 @aidan-kwon 
That functions exists Ethereum console also, but not able to use.
Ethereum drop all `eth_compile*` actions in https://github.com/ethereum/go-ethereum/pull/3740 PR, but still exists in console. 
All these apis is not working currently on geth console too, so I think we deprecating(= removing) them looks good.

jimni1222 replied:
I removed compile apis. Please check https://github.com/klaytn/klaytn/pull/1077/commits/eafaa2bb7e2615c13f033b6b58c605d6a155b615

aidan-kwon commented on console/jsre/deps/web3.js:
```diff
@@ -5626,6 +5628,362 @@ Klay.prototype.isSyncing = function (callback) {
+        outputFormatter: utils.toDecimal
+    });
+
+    var compileSolidity = new Method({
+        name: 'compile.solidity',
+        call: 'eth_compileSolidity',
+        params: 1
+    });
+
+    var compileLLL = new Method({
+        name: 'compile.lll',
+        call: 'eth_compileLLL',
+        params: 1
+    });
+
+    var compileSerpent = new Method({
+        name: 'compile.serpent',
+        call: 'eth_compileSerpent',
+        params: 1
+    });
+
```
same here

jimni1222 replied:
I removed compile apis. Please check https://github.com/klaytn/klaytn/pull/1077/commits/eafaa2bb7e2615c13f033b6b58c605d6a155b615

aeharvlee reviewed (approved):
LGTM 👍

yoomee1313 commented:
@hqjang-pepper please take a look
